File: codechef/FLOW016.py
# Eucliean algorithm for GCD
import fractions 
T = int(raw_input())

for i in range(T):
    a, b = map(int, raw_input().split())
    '''
    temp_b = b
    temp_a = a
    while temp_b:
        temp = temp_b
        temp_b = temp_a % temp_b
        temp_a = temp
    res = temp_a
    '''
    res = fractions.gcd(a, b)
    lcm = (a*b)/res
    print '{} {}'.format(res, lcm)
    

File: codechef/FLOW016.c
#include <stdio.h>

int main() {
    unsigned long long T, A, B, tA, tB, i, temp, gcd, lcm;
    scanf("%llu", &T);
    for(i=0; i<T; i++) {
        scanf("%llu %llu", &A, &B);
        tA = A;
        tB = B;
        while(tB != tA) {
            if (tA > tB)
                tA = tA - tB;
            else 
                tB = tB - tA;
        }
        gcd = tA;
        lcm = (A*B)/gcd;
        printf("%llu %llu\n", gcd, lcm);
    }
    return 0;
}
